scripts/lib/gh-hook-parser.coffee

```diff
@@ -0,0 +1,32 @@
+crypto = require("crypto")
+
+GITHUB_WEBHOOK_SECRET = process.env.GITHUB_WEBHOOK_SECRET
+
+ghHookParser =
+  parseHookRequest: (request) ->
+    eventType = request.headers["x-github-event"]
+    this[eventType]?(request.body) if this.hasValidSignature(request)
+
+  hasValidSignature: (request) ->
+    github_signature = request.headers["x-hub-signature"]
+    payload = JSON.stringify(request.body)
+    sig = "sha1=" + crypto.createHmac("sha1", GITHUB_WEBHOOK_SECRET).update(payload).digest("hex")
+    github_signature is sig
+
+  status: (data) ->
+    {
+      eventType: "status"
+      user: data["commit"]["author"]["login"]
```

Comment:
Question: the `data["commit"]["author"]["login"]` is the author of the commit or the author of the PR?

Reply:
Of the commit. On github the checks are per commit (and not per PR). On their PR UI we basically see the status of the last commit PR, but we are not able to query (or in this case receive) status per PR